src/components/KeyboardShortcutsModal.js

```diff
@@ -108,77 +70,113 @@ class KeyboardShortcutsModal extends React.Component {
+            },
+            openShortcutModalConfig.descriptionKey,
+            openShortcutModalConfig.modifiers,
+            true,
         );
-    }
-
-    render() {
-        const shortcuts = KeyboardShortcut.getDocumentedShortcuts();
-        const modalType = this.props.isSmallScreenWidth ? CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED : CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE;
-
-        return (
-            <Modal
-                isVisible={this.props.isShortcutsModalOpen}
-                type={modalType}
-                innerContainerStyle={{...styles.keyboardShortcutModalContainer, ...StyleUtils.getKeyboardShortcutsModalWidth(this.props.isSmallScreenWidth)}}
-                onClose={KeyboardShortcutsActions.hideKeyboardShortcutModal}
-            >
-                <HeaderWithBackButton
-                    title={this.props.translate('keyboardShortcutModal.title')}
-                    shouldShowCloseButton
-                    shouldShowBackButton={false}
-                    onCloseButtonPress={KeyboardShortcutsActions.hideKeyboardShortcutModal}
-                />
-                <ScrollView style={[styles.p5, styles.pt0]}>
-                    <Text style={styles.mb5}>{this.props.translate('keyboardShortcutModal.subtitle')}</Text>
-                    <View style={[styles.keyboardShortcutTableWrapper]}>
-                        <View style={[styles.alignItemsCenter, styles.keyboardShortcutTableContainer]}>
-                            {_.map(shortcuts, (shortcut, index) => {
-                                const isFirstRow = index === 0;
-                                return this.renderRow(shortcut, isFirstRow);
-                            })}
-                        </View>
+
+        if (props.isShortcutsModalOpen) {
+            // The modal started open, which can happen if you reload the page when the modal is open.
+            subscribeOpenModalShortcuts();
+        }
+
+        return () => {
+            if (unsubscribeShortcutModal) {
+                unsubscribeShortcutModal();
+            }
+            unsubscribeOpenModalShortcuts();
+        };
+        // We only want this to run on mount
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        if (props.isShortcutsModalOpen) {
+            subscribeOpenModalShortcuts();
+        } else {
+            // Modal is closing, remove keyboard shortcuts
+            unsubscribeOpenModalShortcuts();
+        }
+        // subscribeOpenModalShortcuts and unsubscribeOpenModalShortcuts functions are not added as dependencies since they don't change between renders
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.isShortcutsModalOpen]);
+
+    const shortcuts = KeyboardShortcut.getDocumentedShortcuts();
+    const modalType = props.isSmallScreenWidth ? CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED : CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE;
```

Comment:
will be nice to add variables outside of function and apply for them const variables